### DISCUSSION TITLE
Sonoff Zbbridge EZSP - Flash issue

### DISCUSSION THREAD

**Discussion by neilfx1** [arendst/Tasmota#22681] [Support request]
I accidentally flashed the wrong version of EZSP to my Sonoff Zbbridge (original not pro).  As a result, I get this now on boot:

```
16:20:12.838 ZIG: Resetting EZSP device
16:20:14.125 MQT: tele/zgb_bridge/RESULT = {"ZbState":{"Status":1,"Message":"EFR32 EZSP booted","RestartReason":"Power-on","Code":2}}
16:20:14.127 ZIG: ZbEZSPSend 000008
16:20:14.168 ZIG: {"ZbEZSPReceived":"00000D023074"}
16:20:14.186 MQT: tele/zgb_bridge/RESULT = {"ZbState":{"Status":55,"Version":"7.4.3.0","Protocol":13,"Stack":2}}
16:20:14.217 MQT: tele/zgb_bridge/RESULT = {"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}
16:20:14.254 MQT: tele/zgb_bridge/RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
16:20:14.256 ZIG: Abort
```

I've tried to upload the previously installed version (6.7.8) however no matter how many times I upload it, nothing happens.  It gets this far and then stops:

```
16:38:17.860 HTP: Firmware Upgrade
16:38:22.790 UPL: File ncp-uart-sw_6.7.8_115200.ota (190756 bytes)
16:38:23.056 CFG: Saved to flash at 1FB, Count 549, Bytes 4096
16:38:23.073 UPL: File type 5
16:38:25.583 UPL: Progress 100 KB
16:38:27.583 UPP: Multicast (re)joined
16:38:27.585 UPL: Transfer 190756 bytes
16:38:27.585 UPL: Successful 190756 bytes
16:38:27.597 HTP: Upload transfer
```

Manually restarting:

```
00:00:00.004 UFS: FlashFS mounted with 208 kB free
00:00:00.039 CFG: Loaded from File, Count 549
00:00:00.046 ZIG: ZBBridge EEPROM found at address 0x50
00:00:00.047 SNS: Hardware Serial
00:00:00.048 SRC: Restart
00:00:00.049 Project tasmota - zgb-garage Version 14.4.1.1(b7da75d-zbbridge)-2_7_8(2024-12-18T21:50:40)
00:00:00.049 ZIG: Resetting EZSP device
00:00:00.050 ZIG: Loading Zigbee data
00:00:00.075 ZIG: EEPROM signature 0x6D736154 is correct
00:00:00.078 ZIG: Zigbee device information found in File System (6 devices - 188 bytes)
00:00:00.084 ZIG: Zigbee device data in EEPROM (141 bytes)
00:00:00.135 ZIG: ZbLoad '<internal_plugin>' loaded successfully
```

Which then comes back with the same incompatible version installed message.  I'd like to force update the flash, is there anything I can do here?  I tried the various other suggestions I've seen on here except a factory reset which I'd rather not do if I can.

**Comment by s-hadinger** [comment]
Not sure what you can do here. A factory reset will not restore the factory firmware anyways

**Comment by darek-margas** [comment]
I wonder if there is any plan to support proto 12.

**Reply by s-hadinger** [reply]
What is proto 12?

**Reply by darek-margas** [reply]
Protocol 12. It is working up to 8, then there is 12 and 13 - unsupported.

**Reply by s-hadinger** [reply]
Ah ok. No, there are no plans to support beyond protocol v8. There are breaking changes that would require in-depth testing.

Overall I see more value in CC2652P which kept backwards compatibility since ZStack 3

**Reply by sfromis** [reply]
One option could be to use TCP tunneling (ser2net) connecting to Z2M or ZHA, then protocol support would be up to those systems, with Tasmota not being involved in Zigbee handling.

**Reply by darek-margas** [reply]
CC sems a bit worse and the only bridge which has ethernet has FR.
BTW - I ended up in the same situation with inability to downgrade coordinator FW. Thought about recompiling tasmota but tested serial and this is piece of cake, serial over tcp  does work, zha is happy (just tested it) so I tried to downgrade by serial. Havne't found any method to put it into bootloader over TCP serial. Is there a way or, maybe, it can be somehow flashed using tasmota out of incompatible version?
Using Z2M or ZHA isn't really option for me, too much to run and to unstable.

**Reply by s-hadinger** [reply]
> CC sems a bit worse and the only bridge which has ethernet has FR.

There are numerous bridges with Ethernet and CC2652P.

Why dont downgrading work? Does Tasmota refuses to flash or is the firmware rejected by EFR32?

**Reply by darek-margas** [reply]
It is exactly as described above. Normally when Tasmota runs Zigbee, uploading coordinator firmware saves it on filesystem, then tasmota puts coordinator into bootloader and flashes it, Then restarts and viola. However, when it happens as per above, it says "Abort" and stops coordinator, it does not work. Uploading firmware works, it says it has been saved to the filesystem and then nothing happens. No actions follows it hence it is not being flashed into coordinator. Dead end.

**Reply by darek-margas** [reply]
4 of bridges I  now are - 3 x Sonoff, one with ethernet, tube, and that's it. Tube is not really manufactured.

**Comment by hipur** [comment]
I just flashed with this FW http://ota.tasmota.com/tasmota/release/tasmota-zbbridge.bin and having the same issue. 

I followed this tutorial: https://digiblur.com/2020/07/25/how-to-use-the-sonoff-zigbee-bridge-with-home-assistant-tasmota/

This is my console output after 1st reboot:
```
00:00:00.001 HDW: ESP8266EX
00:00:00.005 UFS: FlashFS mounted with 224 kB free
00:00:00.034 CFG: Loaded from File, Count 5
00:00:00.038 QPC: Count 1
00:00:00.043 ZIG: ZBBridge EEPROM found at address 0x50
00:00:00.044 SNS: Hardware Serial
00:00:00.046 Project tasmota - Tasmota Version 14.4.1(release-zbbridge)-2_7_8(2024-12-15T13:29:52)
00:00:00.046 ZIG: Resetting EZSP device
00:00:00.047 ZIG: Loading Zigbee data
00:00:00.112 ZIG: No Zigbee device information
00:00:00.121 ZIG: No Zigbee device data
00:00:00.123 ZIG: ZbLoad '<internal_plugin>' loaded successfully
00:00:01.188 WIF: Connecting to AP1 XXXX in mode 11n as tasmota-04BD1A-7450...
00:00:04.755 WIF: Connected
00:00:05.006 HTP: Web server active on tasmota-04BD1A-7450 with IP address 192.168.1.98
16:05:21.013 RSL: INFO1 = {"Info1":{"Module":"Sonoff ZbBridge","Version":"14.4.1(release-zbbridge)","FallbackTopic":"cmnd/DVES_04BD1A_fb/","GroupTopic":"cmnd/tasmotas/"}}
16:05:21.015 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-04BD1A-7450","IPAddress":"192.168.1.98"}}
16:05:21.016 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":2}}
16:05:21.707 QPC: Reset
16:05:25.503 RSL: STATE = {"Time":"2025-02-12T16:05:25","Uptime":"0T00:00:11","UptimeSec":11,"Vcc":3.427,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"@rodrigoebeta","BSSId":"C0:6F:98:88:E5:E6","Channel":6,"Mode":"11n","RSSI":100,"Signal":-27,"LinkCount":1,"Downtime":"0T00:00:06"}}
16:05:30.152 ZIG: Resetting EZSP device
16:05:30.488 ZbEZSPReceived: time-out, discarding FE0D0180020000000001011060
16:05:40.277 ZIG: timeout, goto label 99
16:05:40.279 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
16:05:40.280 ZIG: Abort
16:05:40.280 ZIG: Stopping (99)
```

Any suggestions on how to be able to communicate with Zigbee module?

**Reply by s-hadinger** [reply]
Please set `weblog 4` and reboot

**Reply by hipur** [reply]
Sorry, didn't read this comment on time.

I was able to flash ZigBee firmware by having 3 windows open. One with the console to monitor, one with the OTA file already selected and one for selecting the module.

After setting the module to Sonoff ZBBridge 75, as soon as it rebooted, I pressed the upload button on the OTA page and firmware was flashed.

However, I now have this in my logs:
```
00:00:04.510 WIF: Connected
00:00:04.763 HTP: Web server active on tasmota-04BD1A-7450 with IP address 192.168.1.98
00:00:05.292 RTC: UTC 2025-02-12T15:35:32Z, DST 2025-03-30T02:00:00, STD 2025-10-26T03:00:00
16:35:32.000 RTC: Synced by NTP
16:35:32.512 RSL: INFO1 = {"Info1":{"Module":"Sonoff ZbBridge","Version":"14.4.1(release-zbbridge)","FallbackTopic":"cmnd/DVES_04BD1A_fb/","GroupTopic":"cmnd/tasmotas/"}}
16:35:32.514 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-04BD1A-7450","IPAddress":"192.168.1.98"}}
16:35:32.515 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":8}}
16:35:34.625 HTP: Main Menu
16:35:35.258 APP: Boot Count 8
16:35:36.415 CFG: Saved to flash at 1FB, Count 19, Bytes 4096
16:35:37.258 RSL: STATE = {"Time":"2025-02-12T16:35:37","Uptime":"0T00:00:12","UptimeSec":12,"Vcc":3.430,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":27,"MqttCount":0,"Wifi":{"AP":1,"SSId":"@rodrigoebeta","BSSId":"C0:6F:98:88:E5:E6","Channel":6,"Mode":"11n","RSSI":100,"Signal":-30,"LinkCount":1,"Downtime":"0T00:00:06"}}
16:35:41.898 ZIG: Resetting EZSP device
16:35:43.169 RSL: RESULT = {"ZbState":{"Status":1,"Message":"EFR32 EZSP booted","RestartReason":"Power-on","Code":2}}
16:35:43.170 ZIG: ZbEZSPSend 000008
16:35:43.222 ZIG: {"ZbEZSPReceived":"00000D025074"}
16:35:43.223 RSL: RESULT = {"ZbState":{"Status":55,"Version":"7.4.5.0","Protocol":13,"Stack":2}}
16:35:43.224 RSL: RESULT = {"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}
16:35:43.226 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
16:35:43.226 ZIG: Abort
16:35:43.227 ZIG: Stopping (99)
```

**Reply by s-hadinger** [reply]
What firmware did you flash?

**Reply by hipur** [reply]
This one:
https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/ZigbeeBridge_SM-011-signed/ncp-uart-sw-7.4.5.0_115200.ota

Since was my first time using Tasmota I was still figuring out how to use it. But after flashing, I just used this command on my console and was able to connect to Z2M on HA.
`backlog rule1 on system#boot do TCPStart 8888 endon ; rule1 1 ; template {"NAME":"Sonoff ZHABridge","GPIO":[56,208,0,209,59,58,0,0,0,0,0,0,17],"FLAG":0,"BASE":18} ; module 0`

**Reply by s-hadinger** [reply]
This firmware is not supported by Tasmota. You should use one of the firmware in the Tasmota repository.

**Reply by sfromis** [reply]
Either that (if you want to use Z2T), or (as you did) connect to Z2M or ZHA to handle the protocol.

**Reply by darek-margas** [reply]
> This firmware is not supported by Tasmota. You should use one of the firmware in the Tasmota repository.

What if after "abort" it does not flash anymore?

**Reply by sfromis** [reply]
Did you try it? If it fails to flash another EZSP firmware version, do post logs of that.

The "Abort" simply is after
`{"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}`
meaning that the Tasmota Zigbee support is not compatible with that firmware version. That does not indicate an *expectation* of becoming unable to flash another EZSP firmware build.

**Reply by darek-margas** [reply]
> Did you try it? If it fails to flash another EZSP firmware version, do post logs of that.
> 
> The "Abort" simply is after `{"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}` meaning that the Tasmota Zigbee support is not compatible with that firmware version. That does not indicate an _expectation_ of becoming unable to flash another EZSP firmware build.

Here it is. It stops there forever. With working version it does flash and does reboot after those upload messages:
`
00:00:00.001 HDW: ESP8266EX

00:00:00.004 UFS: FlashFS mounted with 224 kB free

00:00:00.036 CFG: Loaded from File, Count 5

00:00:00.039 QPC: Count 1

00:00:00.045 ZIG: ZBBridge EEPROM found at address 0x50

00:00:00.046 SNS: Hardware Serial

00:00:00.047 Project tasmota - Tasmota Version 14.4.1(release-zbbridge)-2_7_8(2024-12-15T13:29:52)

00:00:00.048 ZIG: Resetting EZSP device

00:00:00.048 ZIG: Loading Zigbee data

00:00:00.114 ZIG: No Zigbee device information

00:00:00.119 ZIG: Zigbee device data in EEPROM (129 bytes)

00:00:00.163 ZIG: ZbLoad '<internal_plugin>' loaded successfully

00:00:01.199 WIF: Connecting to AP1 IOT in mode 11n as tasmota-9E30EB-4331...

00:00:04.752 WIF: Connected

00:00:05.003 HTP: Web server active on tasmota-9E30EB-4331 with IP address 192.168.16.120

14:08:59.010 RSL: INFO1 = {"Info1":{"Module":"Sonoff ZbBridge","Version":"14.4.1(release-zbbridge)","FallbackTopic":"cmnd/
DVES_9E30EB_fb/","GroupTopic":"cmnd/tasmotas/"}}

14:08:59.011 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-9E30EB-4331","IPAddress":"192.168.16.120"}}

14:08:59.012 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":2}}

14:08:59.643 QPC: Reset

14:09:03.500 RSL: STATE = {"Time":"2025-02-13T14:09:03","Uptime":"0T00:00:12","UptimeSec":12,"Vcc":3.471,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"IOT","BSSId":"F4:8C:EB:86:83:31","Channel":9,"Mode":"11n","RSSI":100,"Signal":-39,"LinkCount":1,"Downtime":"0T00:00:06"}}

14:09:08.111 ZIG: Resetting EZSP device

14:09:09.366 RSL: RESULT = {"ZbState":{"Status":1,"Message":"EFR32 EZSP booted","RestartReason":"Power-on","Code":2}}

14:09:09.421 RSL: RESULT = {"ZbState":{"Status":55,"Version":"7.3.1.0","Protocol":12,"Stack":2}}

14:09:09.422 RSL: RESULT = {"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}

14:09:09.424 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}

14:09:09.424 ZIG: Abort

14:09:09.425 ZIG: Stopping (99)

14:10:12.020 UPL: File ncp-uart-nsw_6.7.9_115200.ota (192240 bytes)

14:10:14.453 UPP: Multicast (re)joined

14:10:14.454 UPL: Transfer 192240 bytes

14:10:14.454 UPL: Successful 192240 bytes

`

**Reply by hipur** [reply]
Even though Tasmota does not support the newest Zb module firmware, it is still fully functional to use with Z2M. If you need to flash another firmware, just do so during the 5 seconds after booting while Tasmota is still communicating with the module.

**Reply by darek-margas** [reply]
with weblog  4
`
00:33:19.856 HTP: Firmware Upgrade

00:33:23.740 WIF: Checking connection...

00:33:32.461 UPL: File ncp-uart-nsw_6.7.9_115200.ota (192240 bytes)

00:33:32.559 CFG: Saved to flash at 1FB, Count 22, Bytes 4096

00:33:32.576 UPL: File type 5

00:33:34.540 UPL: Progress 100 KB

00:33:36.189 UPP: Multicast (re)joined

00:33:36.190 UPL: Transfer 192240 bytes

00:33:36.191 UPL: Successful 192240 bytes

00:33:36.203 HTP: Upload transfer`

**Comment by darek-margas** [comment]
I created bug report for this (#23001) and I explained there how I was able to recover from incompatible coordinator firmware

**Reply by Jason2866** [reply]
closed this issue, flashing other firmwares than the provided in the repo bricks the device.
Coordinator firmware needs specific changes to be compatible. When not changed, the device is no more flashable when a not modified (compatible) version was flashed.
Only low level flashing with an seeger JTAG Flasher is possible in this case.
